osu.Game.Rulesets.Taiko/UI/DrawableTaikoMascot.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Audio.Track;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Textures;
+using osu.Game.Beatmaps.ControlPoints;
+using osu.Game.Graphics.Containers;
+
+namespace osu.Game.Rulesets.Taiko.UI
+{
+    public sealed class DrawableTaikoMascot : BeatSyncedContainer
+    {
+        private static TaikoMascotTextureAnimation idleDrawable, clearDrawable, kiaiDrawable, failDrawable;
+        private EffectControlPoint lastEffectControlPoint;
+        private TaikoMascotAnimationState state;
+
+        public Bindable<TaikoMascotAnimationState> PlayfieldState;
+
+        /// <summary>
+        /// Determines if there should be no "state logic", intended for testing.
+        /// </summary>
+        public bool Dumb { get; set; }
```

Comment:
This ideally shouldn't be a thing, much less be named this way and be `public`. I'd make it `protected` at the least, or better yet, make `getFinalAnimationState` `protected virtual`, subclass the mascot in the test scene and override that method to make it a no-op.